lib/console.js

```diff
@@ -9,25 +9,24 @@ function Console(stdout, stderr) {
   if (!stdout || typeof stdout.write !== 'function') {
     throw new TypeError('Console expects a writable stream instance');
   }
-  if (!stderr) {
-    stderr = stdout;
-  }
-  var prop = {
+  const prop = {
     writable: true,
     enumerable: false,
-    configurable: true
+    configurable: true,
+    value: stdout
   };
-  prop.value = stdout;
   Object.defineProperty(this, '_stdout', prop);
-  prop.value = stderr;
+  if (stderr) {
+    prop.value = stderr;
+  }
   Object.defineProperty(this, '_stderr', prop);
   prop.value = new Map();
   Object.defineProperty(this, '_times', prop);
 
   // bind the prototype functions to this Console instance
-  var keys = Object.keys(Console.prototype);
-  for (var v = 0; v < keys.length; v++) {
-    var k = keys[v];
+  const keys = Object.keys(Console.prototype);
+  for (let v = 0; v < keys.length; v++) {
```

Comment:
Will check with IRHydra today, if it will make sense. Compiled d8 from
latest source some days ago.

On Wed, Jan 27, 2016, 06:28 Ron Korving <notifications@github.com> wrote:

> In lib/console.js
> <https://github.com/nodejs/node/pull/4880#discussion_r50937364>:
>
> >    Object.defineProperty(this, '_stderr', prop);
> >    prop.value = new Map();
> >    Object.defineProperty(this, '_times', prop);
> >
> >    // bind the prototype functions to this Console instance
> > -  var keys = Object.keys(Console.prototype);
> > -  for (var v = 0; v < keys.length; v++) {
> > -    var k = keys[v];
> > +  const keys = Object.keys(Console.prototype);
> > +  for (let v = 0; v < keys.length; v++) {
>
> There is a function call (bind) happening though, so V8 cannot optimize
> away the length lookup (according to @mraleph <https://github.com/mraleph>
> in that article). Now that's more than a year old, and it's possible that
> special built-in functions like bind don't tamper with optimization. Also,
> if bind gets inlined, length lookups should be optimized (but who knows).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nodejs/node/pull/4880/files#r50937364>.
>
-- 
Sincerely,
Alexander Makarenko